src/main/java/frc/robot/systems/BallHandlingFSM.java

```diff
@@ -9,25 +11,39 @@
 
 // Robot Imports
 import frc.robot.TeleopInput;
+import frc.robot.wrappers.NeoSparkMaxPid;
 import frc.robot.HardwareMap;
 
 public class BallHandlingFSM {
 	/* ======================== Constants ======================== */
 	// FSM state definitions
 	public enum FSMState {
 		IDLE,
-		FIRING
+		FIRING,
+		RETRACTING,
+		INTAKING,
+		RELEASING
 	}
 
-	private static final int PUSH_TIME_SECONDS = 3;
+	private static final double PUSH_TIME_SECONDS = 1;
+
+	private static final double INTAKE_MOTOR_RPM = 700;
 
 	/* ======================== Private variables ======================== */
 	private FSMState currentState;
 
 	private Solenoid pushSolenoid;
+	private Solenoid pullSolenoid;
```

Comment:
You probably are looking for DoubleSolenoid https://first.wpi.edu/wpilib/allwpilib/docs/release/java/edu/wpi/first/wpilibj/DoubleSolenoid.html

Reply:
Implemented